spec/models/reporting_pipeline/patient_states_per_month_spec.rb

```diff
@@ -5,6 +5,17 @@
     it { should belong_to(:patient) }
   end
 
+  around do |example|
+    # We need to enforce a known time for this test, otherwise we will have intermittent failures. For example,
+    # if we use live system time, many of these specs will fail after 18:30 UTC (ie 14:30 ET) when on the last day of a month,
+    # because that falls into the next day in IST (our reporting time zone). So to prevent confusing failures for
+    # developers or CI during North American afternoons, we freeze to a time that will be the end of the month for
+    # UTC, ET, and IST. Timezones! 🤯
+    Timecop.freeze("June 30 2021 5:30 UTC") do # June 30th 23:00 IST time
+      example.run
+    end
+  end
```

Comment:
The issue here is in using relative times. This was always the case, perhaps the reporting tables make the issue more apparent.

My preference would be to use the absolute times for the test as in `june_2021` that we setup in the helper, as opposed to relative times.

Reply:
> My preference would be to use the absolute times for the test as in `june_2021` that we setup in the helper, as opposed to relative times.

Yeah, I definitely prefer absolute times as well. I think the point I was trying to make is that even with the `june_2021` helper, we need to establish an absolute time via `freeze`, or things will fail like this intermittently.  

This spec _does_ use `june_2021` helper btw, but that doesn't help when time is a frustrating continually advancing thing.  😃

Reply:
Hrm, I'm not sure I follow. What are the timestamps in concern that are advancing? Is it the time at which refresh runs that's the issue?

Reply:
> Hrm, I'm not sure I follow. What are the timestamps in concern that are advancing? Is it the time at which refresh runs that's the issue?

The timestamps that are advancing is system time, i.e. `Time.current` ...without the `Timecop.freeze`, we would have intermittent failures in this spec depending on the date and time the spec is being run.